env/io_posix.cc

```diff
@@ -371,7 +371,7 @@ size_t PosixHelper::GetUniqueIdFromFile(int fd, char* id, size_t max_size) {
 }
 #endif
 
-#ifdef OS_LINUX
+#if defined(OS_LINUX) || defined(OS_MACOSX)
```

Comment:
What about the #ifdef OS_LINUX around line 468?  Doesn't this one need changed as well?

Reply:
> Also, shouldn't the code in fs_posix.cc that creates and uses a cache be changed to use macOS as well?

@mrambacher Can you be a bit more specific... do you mean `logical_block_size_cache_`?

Reply:
Yes, I meant the logical_block_size_cache_.

If it turns out a bunch of places are using this same defined() rule, it might be worth introducing a define ROCKSDB_SUPPORTS_BLOCK_SIZE (or some other descriptive name) define.  Define it in io_posix.h and use it whenever the check is needed.